module/platform/Webots/src/Webots.cpp

```diff
@@ -0,0 +1,228 @@
+/*
+ * This file is part of the NUbots Codebase.
+ *
+ * The NUbots Codebase is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * The NUbots Codebase is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with the NUbots Codebase.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright 2021 NUbots <nubots@nubots.net>
+ */
+
+#include "Webots.hpp"
+
+#include <fmt/format.h>
+
+#include "extension/Configuration.hpp"
+
+#include "message/motion/ServoTarget.hpp"
+#include "message/output/CompressedImage.hpp"
+#include "message/platform/darwin/DarwinSensors.hpp"
+#include "message/platform/webots/ConnectRequest.hpp"
+#include "message/platform/webots/messages.hpp"
+
+// Include headers needed for TCP connection
+#include <netdb.h>      /* definition of gethostbyname */
+#include <netinet/in.h> /* definition of struct sockaddr_in */
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <unistd.h> /* definition of close */
+
+namespace module::platform {
+
+using extension::Configuration;
+using message::motion::ServoTargets;
+using message::output::CompressedImage;
+using message::platform::darwin::DarwinSensors;
+
+using message::platform::webots::ActuatorRequests;
+using message::platform::webots::ConnectRequest;
+using message::platform::webots::MotorPosition;
+using message::platform::webots::MotorTorque;
+using message::platform::webots::MotorVelocity;
+using message::platform::webots::SensorMeasurements;
+
+int Webots::tcpip_connect(const std::string& server_name, const int& port) {
+    // Create the socket
+    int fd = socket(AF_INET, SOCK_STREAM, 0);
+
+    // Store the ip address information that we will connect to
+    sockaddr_in address;
+    memset(&address, 0, sizeof(sockaddr_in));
+    address.sin_family = AF_INET;
+    address.sin_port   = htons(port);
+    hostent* server    = gethostbyname(server_name.c_str());
```

Comment:
the [linux docs](https://man7.org/linux/man-pages/man3/gethostbyname.3.html#DESCRIPTION) say you should use `getaddrinfo(..)` instead of `gethostbyname(..)`. I'll leave that for you to decide if that's worth acting on.

Reply:
Yea, we should avoid using deprecated functions.